### PR TITLE
Added hook to be able to modify assets cache busting strategy

### PR DESCRIPTION
**Pull request by fabioperrella**
Added hook to be able to modify assets cache busting strategy using a plugin

refering to #5397

### CODE REVIEW

File: program/include/rcmail_output_html.php [hunk 1]
```diff
@@ -852,17 +852,30 @@ protected function file_mod($file)
...
         return $file;
     }
 
+    private function file_with_cache_buster($file_name, $file_timestamp){
+        $hook_response = $this->app->plugins->exec_hook('file_with_cache_buster', array(
+            'file_name' => $file_name,
+            'file_timestamp' => $file_timestamp
+        ));
+
+        if (isset($hook_response['file_with_cache_buster']) && $hook_response['file_with_cache_buster']){
```
Comment (AllanKlaus) [inline program/include/rcmail_output_html.php]: You could use !empty($hook_response['file_with_cache_buster']) because without it php can launch notice

**Comment by thomascube** [comment]
Looks good to me. @alecpl any objections?

File: program/include/rcmail_output_html.php [hunk 2]
```diff
@@ -852,17 +852,30 @@ protected function file_mod($file)
...
+            $file = $this->file_with_cache_buster($file, $fs);
         }
 
         return $file;
     }
 
+    private function file_with_cache_buster($file_name, $file_timestamp){
+        $hook_response = $this->app->plugins->exec_hook('file_with_cache_buster', array(
+            'file_name' => $file_name,
+            'file_timestamp' => $file_timestamp
```
Comment (alecpl) [inline program/include/rcmail_output_html.php]: I'd change file_name and file_timestamp to just name and timestamp.

File: program/include/rcmail_output_html.php [hunk 3]
```diff
@@ -852,17 +852,30 @@ protected function file_mod($file)
...
+        ));
+
+        if (isset($hook_response['file_with_cache_buster']) && !empty($hook_response['file_with_cache_buster'])){
+            return $hook_response['file_with_cache_buster'];
```
Comment (alecpl) [inline program/include/rcmail_output_html.php]: s/file_with_cache_buster/result/

Comment (alecpl) [inline program/include/rcmail_output_html.php]: and isset() is not needed, !empty() is enough.

File: program/include/rcmail_output_html.php [hunk 4]
```diff
@@ -852,17 +852,30 @@ protected function file_mod($file)
...
         if ($fs = @filemtime($this->assets_dir . $file)) {
-            $file .= '?s=' . $fs;
+            $file = $this->file_with_cache_buster($file, $fs);
         }
 
         return $file;
     }
 
+    private function file_with_cache_buster($file_name, $file_timestamp){
+        $hook_response = $this->app->plugins->exec_hook('file_with_cache_buster', array(
```
Comment (alecpl) [inline program/include/rcmail_output_html.php]: I don't like the hook name. How about template-file-cache-buster?

**Pull request closed without merging**